apps/cowswap-frontend/src/common/hooks/useAnnouncements.ts

```diff
@@ -0,0 +1,64 @@
+import { useAtomValue } from 'jotai'
+import { useMemo } from 'react'
+
+import { isProdLike } from '@cowprotocol/common-utils'
+import { Announcements, announcementsAtom } from '@cowprotocol/core'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+
+function useAnnouncements(chainId: SupportedChainId): Announcements {
+  const allAnnouncements = useAtomValue(announcementsAtom)
+
+  return useMemo(() => {
```

Comment:
You probably not need to use `useMemo` anymore in cases like that

Reply:
Would be amazing to add some console logs and assert this to see how smarter the compiler had gotten

Reply:
Nope, seems to do a lot of re-renders without useMemo, I'll keep it for now.